api/specs/web-server/_common.py

```diff
@@ -8,15 +8,70 @@
 from typing import Any, ClassVar, NamedTuple
 
 import yaml
-from fastapi import FastAPI
+from fastapi import FastAPI, Query
 from models_library.basic_types import LogLevel
-from pydantic import BaseModel, Field
+from models_library.utils.json_serialization import json_dumps
+from pydantic import BaseModel, Field, create_model
 from pydantic.fields import FieldInfo
 from servicelib.fastapi.openapi import override_fastapi_openapi_method
 
 CURRENT_DIR = Path(sys.argv[0] if __name__ == "__main__" else __file__).resolve().parent
 
 
+def _create_json_type(**schema_extras):
+    class _Json(str):
+        __slots__ = ()
+
+        @classmethod
+        def __modify_schema__(cls, field_schema: dict[str, Any]) -> None:
+            # openapi.json schema is corrected here
+            field_schema.update(
+                type="string",
+                # format="json-string" NOTE: we need to get rid of openapi-core in web-server before using this!
+            )
+            if schema_extras:
+                field_schema.update(schema_extras)
+
+    return _Json
+
+
+def as_query(model_class: type[BaseModel]) -> type[BaseModel]:
+    fields = {}
```

Comment:
this will be a funny migration job to pyV2

Reply:
Let's see. i am optimistic here.